src/components/Forms/JoinTeamForm.vue

```diff
@@ -164,9 +164,34 @@ export default {
 		handleRejectTeam() {
 			this.showError = false;
 			window.location.href = `/declineInvitationToJoinTeam?team_id=${this.teamId}&doneUrl=${this.doneUrl}`;
+		},
+		handleAlreadyJoined() {
+			const teamId = this.$route.query.team_id ? numeral(this.$route.query.team_id).value() : null;
+			if (teamId) {
+				this.loading = true;
+				this.apollo.query({
+					query: myTeamsQuery,
+					variables: {
+						teamIds: [teamId]
+					}
+				}).then(({ data }) => {
+					this.isMember = _get(data, 'my.teams.values', []).length;
+					// if lender is a member proceed
+					if (this.isMember) {
+						window.location = '/authenticate/redirect&team_id=278842&promo_id=286';
+					}
+					if (!this.isMember) {
+						this.loading = false;
+					}
+				});
+			}
 		}
 	},
 	created() {
+		// in the browser check if the lender has already joined the team
+		if (typeof window !== 'undefined') {
+			this.handleAlreadyJoined();
```

Comment:
What if we did this in the preFetch instead? That way we could redirect back without having to load the page in the browser (by rejecting the prefetch promise with the url to redirect to)